src/main/java/org/opensearch/performanceanalyzer/PerformanceAnalyzerApp.java

```diff
@@ -250,23 +254,71 @@ private static void startReaderThread(
-                                    }
+                                    retryAttemptLeft--;
                                     LOG.error(
-                                            "Error in ReaderMetricsProcessor...restarting, ExceptionCode: {}",
-                                            StatExceptionCode.READER_RESTART_PROCESSING.toString());
+                                            "Error in ReaderMetricsProcessor...restarting, retryCount left: {}."
+                                                    + "Exception: {}",
+                                            retryAttemptLeft,
+                                            e.getMessage());
                                     StatsCollector.instance()
                                             .logException(
                                                     StatExceptionCode.READER_RESTART_PROCESSING);
+
+                                    if (TroubleshootingConfig.getEnableDevAssert()) break;
+
+                                    // All retry attempts exhausted; handle thread failure
+                                    if (retryAttemptLeft <= 0) handleReaderThreadFailed();
                                 }
                             }
                         },
                         PerformanceAnalyzerThreads.PA_READER);
-
         readerThread.start();
     }
 
+    private static void handleReaderThreadFailed() {
+        // Reader subcomponent is responsible for processing, cleaning metrics written by PA.
+        // Since Reader thread fails to start successfully, execute following:
+        //
+        // 1. Disable PA - Stop collecting OpenSearch metrics
+        // 2. Terminate RCA Process - Gracefully shutdown all
+        //    existing resources/channels, including Reader.
+        try {
+            LOG.info(
+                    "Exhausted {} attempts - unable to start Reader Thread successfully; disable PA",
+                    READER_RESTART_MAX_ATTEMPTS);
+            disablePA();
+            LOG.info("Attempt to disable PA succeeded.");
+            StatsCollector.instance()
+                    .logException(StatExceptionCode.READER_ERROR_PA_DISABLE_SUCCESS);
+        } catch (Throwable e) {
+            LOG.info("Attempt to disable PA failing: {}", e.getMessage());
+            StatsCollector.instance()
+                    .logException(StatExceptionCode.READER_ERROR_PA_DISABLE_FAILED);
+        } finally {
+            cleanupAndExit();
+        }
+    }
+
+    private static void disablePA() {
+        String PA_CONFIG_PATH = Util.PA_BASE_URL + "/cluster/config";
+        String PA_DISABLE_PAYLOAD = "{\"enabled\": false}";
+
+        int resCode =
+                ESLocalhostConnection.makeHttpRequest(
+                        PA_CONFIG_PATH, HttpMethod.POST, PA_DISABLE_PAYLOAD);
+        if (resCode != HttpURLConnection.HTTP_OK) {
+            throw new RuntimeException("Failed to disable PA");
```

Comment:
should we do retry for disabling PA as well :) ?

Reply:
Yes, this can be re-tried.  I will add the re-try attempts here.

I am considering PA disable as a best-effort exercise - PA metric directory `(tmpfs)` is restricted by storage limit so in case RCA process has died, but PA is still running and producing metrics, it will never breach the tmpfs size limit.